### DISCUSSION TITLE
Custom firmware benefits?

### DISCUSSION THREAD

**Discussion by Madddcow** [ludwig-v/wireless-carplay-dongle-reverse-engineering#51] [General]
What are the benefits of the custom firmwares?  I've seen the note below but what practical benefit does the CPU performance scaling governor provide? Or saving RAM by disabling logging?  I assume these factors increase the responsiveness but is it a noticeable improvement?

_"All custom firmwares restore "performance" scaling governor of the CPU, disable logging to save RAM and allow custom script execution from the USB Key"_ 

Cheers
Gary

**Comment by Madddcow** [comment]
No-one?

**Reply by ghost** [reply]
1) cpu performance
2) disable logging which can use a lot of ram
3) allow custom scripts to run 

3 is important because you can use scripts to roll back to a old version or run a script to allow you to ssh/SFTP into the file system. 


currently some people NOT running on custom firmware have updated to STOCK firmware (march 2021) and the new firmware has incompatible Bluetooth/wifi drivers and their devices do not work and are bricked and they cannot downgrade. If they were on a custom firmware they could easily downgrade 

the manufacturer is doing nothing to help. So, custom is better.

**Comment by CiNcH83** [comment]
Interesting. I read somewhere that the adapter might suffer from a heat problem and that installing a heat sink mitigated crashing. The performance governor might be counterproductive in that case. Do you actually feel a difference? What's the original governor setting?
